packages/git/src/common/git.ts

```diff
@@ -672,4 +673,14 @@ export namespace GitUtils {
         return GitFileStatus.Modified;
     }
 
+    /**
+     * Returns the repository relative path of the given uri.
+     * @param repository
+     * @param uri
+     */
+    export function getRepositoryRelativePath(repository: Repository, uri: URI) {
+        const repositoryUri = new URI(repository.localUri);
+        return uri.toString().substr(repositoryUri.toString().length + 1);
```

Comment:
What would be the expected behavior if the URI is not contained in the repository?

Reply:
Since you have moved this method into a utility class, I would not ignore the above-mentioned case silently. What do you think?

Reply:
You are right. So, I moved it into the GitBaseWidget and made it protected. It is only useful for the widgets anyway.